mod/wall_attach.php

```diff
@@ -36,15 +37,13 @@ function wall_attach_post(App $a) {
 		$owner = User::getOwnerDataByNick($nick);
 		if (!DBA::isResult($owner)) {
 			if ($r_json) {
-				echo json_encode(['error' => DI::l10n()->t('Invalid request.')]);
-				exit();
+				System::jsonExit(['error' => DI::l10n()->t('Invalid request.')]);
```

Comment:
Please use `jsonError` for JSON error responses.

Reply:
We don't return any HTTP Error code here. I don't want to change the general behaviour here, since I can't really say if this could have got some side effects.

Reply:
Hmm, we really should return non-200 error codes in these cases though.

Reply:
Surely this is something that we have to check in the future.